File: js/nova/svg/SvgCircle.js
var SvgCircle = function () {
	var self = this;
	
	this.x = 0;
	this.y = 0;
	this.r = 0;
	
	this.SvgCircle = function (x, y, r) {
		self.this(x, y, r);
	};
	
	this.destroy = function () {
	};
	
	this.this = function (x, y, r) {
		self = x;
		self = y;
		self = r;
	};
	
	this.generateOutput = function (file) {
		file.write(new String("<circle cx=\"").concat(Double.toString(self.x).concat(new String("\" cy=\"").concat(Double.toString(self.y).concat(new String("\" r=\"").concat(Int.toString(self.r).concat(new String("\" stroke=\"false\" fill=\"black\"/>\n"))))))));
	};
	
	this.toString = function () {
		return concat(Double.toString(self.x).concat(new String(", ").concat(Double.toString(self.y).concat(new String(") with a radius of ").concat(Int.toString(self.r).concat(new String("]")))))));
};

this.super = function () {
};

};

File: java/nova/web/svg/SvgCircle.java
package nova.web.svg;

import java.util.Optional;
import nova.exception.ExceptionData;
import nova.exception.Exception;
import nova.exception.DivideByZeroException;
import nova.io.NovaConsole;
import nova.primitive.number.Number;
import nova.primitive.number.NovaByte;
import nova.primitive.number.NovaShort;
import nova.primitive.number.NovaInt;
import nova.primitive.number.NovaLong;
import nova.primitive.number.NovaFloat;
import nova.primitive.number.NovaDouble;
import nova.primitive.Null;
import nova.primitive.number.Char;
import nova.primitive.Bool;
import nova.datastruct.list.NovaArray;
import nova.datastruct.list.IntArray;
import nova.datastruct.list.CharArray;
import nova.datastruct.list.DoubleArray;
import nova.datastruct.list.IntRange;
import nova.thread.Thread;
import nova.thread.async.Async;
import nova.gc.GC;
import nova.math.NovaMath;
import nova.NovaObject;
import nova.NovaString;
import nova.System;
import nova.Class;
import nova.io.File;
import nova.web.svg.SvgComponent;

public class SvgCircle extends SvgComponent
{
	
	public double x;
	public double y;
	public int r;
	
	
	
	
	public SvgCircle(double x, double y, int r)
	{
		init(x, y, r);
	}
	
	public void init(double x, double y, int r)
	{
		this.x = x;
		this.y = y;
		this.r = r;
	}
	
	public void generateOutput(File file)
	{
		file.write(new NovaString("<circle cx=\"").concat(NovaDouble.toString((x)).concat(new NovaString("\" cy=\"").concat(NovaDouble.toString((y)).concat(new NovaString("\" r=\"").concat(NovaInt.toString((r)).concat(new NovaString("\" stroke=\"false\" fill=\"black\"/>\n"))))))));
	}
	
	public NovaString toString()
	{
		return new NovaString("[Circle at (").concat(NovaDouble.toString((x)).concat(new NovaString(", ").concat(NovaDouble.toString((y)).concat(new NovaString(") with a radius of ").concat(NovaInt.toString((r)).concat(new NovaString("]")))))));
	}
	
}
